backend/danswer/connectors/xenforo/connector.py

```diff
@@ -0,0 +1,208 @@
+        logger.error(f"Error on {url}")
+        logger.exception(e)
+        pass
+    return BeautifulSoup("", "html.parser")
+
+
+def get_title(soup):
+    title = soup.find("h1", "p-title-value").text
+    for char in (";", ":", "!", "*", "/", "\\", "?", '"', "<", ">", "|"):
+        title = title.replace(char, "_")
+    return title
+
+
+def get_threads(self, url):
+    soup = requestsite(self, url)
+    thread_tags = soup.find_all(class_="structItem-title")
+    base_url = "{uri.scheme}://{uri.netloc}".format(uri=urlparse(url))
+    threads = []
+    for x in thread_tags:
+        y = x.find_all(href=True)
+        for element in y:
+            link = element["href"]
+            if "threads/" in link:
+                stripped = link[0: link.rfind("/") + 1]
+                if base_url + stripped not in threads:
+                    threads.append(base_url + stripped)
+    return threads
+
+
+def get_pages(soup, url):
+    page_tags = soup.select("li.pageNav-page")
+    page_numbers = []
+    for button in page_tags:
+        if re.match("^\d+$", button.text):
+            page_numbers.append(button.text)
+
+    try:
+        max_pages = max(page_numbers, key=int)
+    except ValueError:
+        max_pages = 1
+
+    all_pages = []
+    for x in range(1, int(max_pages) + 1):
+        all_pages.append(f"{url}page-{x}")
+    return all_pages
+
+
+def parse_post_date(post_element: BeautifulSoup):
+    date_string = post_element.find('time')['datetime']
+    post_date = datetime.strptime(date_string, '%Y-%m-%dT%H:%M:%S%z')
+    return datetime_to_utc(post_date)
+
+
+def scrape_page_posts(soup, url, initial_run, start_time) -> list:
+    title = get_title(soup)
+
+    documents = []
+    for post in soup.find_all("div", class_="message-inner"):
+        post_date = parse_post_date(post)
+        if initial_run or post_date > start_time:
+            post_text = post.find("div", class_="bbWrapper").get_text(strip=True) + "\n"
+            author_tag = post.find("a", class_="username")
+            if author_tag is None:
+                author_tag = post.find("span", class_="username")
+            author = author_tag.get_text(strip=True) if author_tag else "Deleted author"
+            document = Document(
+                id=f"{DocumentSource.XENFORO.value}__{title}",
+                sections=[Section(link=url, text=post_text)],
+                title=title,
+                text=post_text,
+                source=DocumentSource.WEB,
+                semantic_identifier=title,
+                primary_owners=[BasicExpertInfo(display_name=author)],
+                metadata={"type": "post", "author": author, "time": post_date.strftime('%Y-%m-%d %H:%M:%S')},
+                doc_updated_at=post_date,
+            )
+            documents.append(document)
+    return documents
+
+
+class XenforoConnector(LoadConnector):
+
+    # Class variable to track if the connector has been run before
+    has_been_run_before = False
```

Comment:
@Weves this connector has to be run through all pages only once and every subsequent run should compare post time with current connector refresh start time (updating once a day). Comparison is here: if initial_run or post_date > start_time. 
question: How can I make sure this happens? unfortunately has_been_run_before is being re-initialized with every new index attempt